mobile/src/main/java/ch/epfl/sdp/mobile/state/StatefulSettingsScreen.kt

```diff
@@ -9,12 +10,14 @@ import ch.epfl.sdp.mobile.ui.setting.SettingScreenState
 import ch.epfl.sdp.mobile.ui.setting.SettingsScreen
 import ch.epfl.sdp.mobile.ui.social.ChessMatch
 
-class AuthenticatedUserProfileScreenState(private val user: AuthenticatedUser) :
-    SettingScreenState {
+class AuthenticatedUserProfileScreenState(
+    private val user: AuthenticatedUser,
+    matches: List<ChessMatch>
+) : SettingScreenState {
   override val email = user.email
-  override val pastGamesCount = 0
+  override val pastGamesCount = matches.size
   override val puzzlesCount = 0
-  override val matches = emptyList<ChessMatch>()
+  override val matches = matches
```

Comment:
The `pastGameCount` and `matches` values won't be updated, since they're passed once to the `AuthenticatedUserProfileScreenState` which is `remember`ed.

There are multiple approaches to implementing a ViewModel, and I suggest following a pattern similar to what you can find here : https://github.com/epfl-SDP/android/blob/main/mobile/src/main/java/ch/epfl/sdp/mobile/state/StatefulGameScreen.kt#L99-L127.

Notice how the `Flow` are collected in the ViewModel, using a `CoroutineScope` that's tied to the lifecycle of the composition. When a new element is collected, the associated `MutableState` is updated.